Checkout/Checkout/Source/Validation/Validators/CardNumberValidator.swift

```diff
@@ -89,12 +89,21 @@ final class CardNumberValidator: CardNumberValidating {
   ) -> Card.Scheme? {
     let range = NSRange(location: 0, length: cardNumber.utf16.count)
 
-    return Card.Scheme.allCases.first { scheme in
+    let matchedScheme = Card.Scheme.allCases.first { scheme in
       cardSchemeToRegex(scheme)?.firstMatch(in: cardNumber, options: [], range: range) != nil
     }
+    return addSchemePropertyIfNeeded(scheme: matchedScheme, cardNumber: cardNumber)
   }
 
   private func validateDigitsOnly(in string: String) -> Bool {
     return CharacterSet.decimalDigits.isSuperset(of: CharacterSet(charactersIn: string))
   }
+    
+  private func addSchemePropertyIfNeeded(scheme: Card.Scheme?, cardNumber: String) -> Card.Scheme? {
+    if scheme == .maestro(length: 0) {
```

Comment:
this is now dependent on the Maestro scheme in `Card.Scheme.allCases` being exactly `.maestro(length: 0)`. We can make this a bit simpler
```suggestion
    if scheme == .maestro {
```

Reply:
That is not possible because the associated type cannot just be ignored and equalise without (see screenshot for error).
<img width="464" alt="Screenshot 2022-08-10 at 12 06 43" src="https://user-images.githubusercontent.com/102028170/183886545-4474ac33-a2b7-4121-a6df-02b0c4fa5bf6.png">
 
 ***
 
 Even if it did work, we would only want to `addSchemePropertyIfNeeded` and if a length existed we wouldn't want to override it.

Reply:
sorry this is only possible in a switch. if you're wondering what that might look like https://github.com/checkout/frames-ios/pull/242/files#r942569571

Reply:
sorry got the syntax a bit wrong here
```suggestion
    if case .maestro = scheme {
```

Reply:
Applied as suggested